tests/amsterdam/eip7928_block_level_access_lists/test_block_access_lists_eip7702.py

```diff
@@ -0,0 +1,526 @@
+    if not self_funded:
+        post.update({relayer: Account(nonce=2)})
+
+    blockchain_test(
+        pre=pre,
+        blocks=[block],
+        post=post,
+    )
+
+
+def test_bal_7702_delegated_storage_access(
+    pre: Alloc,
+    blockchain_test: BlockchainTestFiller,
+):
+    """Ensure BAL captures storage operations when calling a delegated EIP-7702 account."""
+    # Oracle contract that reads from slot 0x01 and writes to slot 0x02
+    oracle = pre.deploy_contract(code=Op.SLOAD(0x01) + Op.PUSH1(0x42) + Op.PUSH1(0x02) + Op.SSTORE)
+    bob = pre.fund_eoa()
+
+    ## Is there a cleaner way to create pre-existing delegation?
+    alice = pre.deploy_contract(nonce=0x1, code=Spec7702.delegation_designation(oracle), balance=0)
+
+    tx = Transaction(
+        sender=bob,
+        to=alice,  # Bob calls Alice (delegated account)
+        value=10,
+        gas_limit=1_000_000,
+        gas_price=0xA,
+    )
+
+    block = Block(
+        txs=[tx],
+        expected_block_access_list=BlockAccessListExpectation(
+            account_expectations={
+                alice: BalAccountExpectation(
+                    balance_changes=[BalBalanceChange(tx_index=1, post_balance=10)],
+                    storage_changes=[
+                        BalStorageSlot(
+                            slot=0x02,
+                            slot_changes=[BalStorageChange(tx_index=1, post_value=0x42)],
+                        )
+                    ],
+                    storage_reads=[0x01],
+                ),
+                bob: BalAccountExpectation(
+                    nonce_changes=[BalNonceChange(tx_index=1, post_nonce=1)],
+                ),
+                # Oracle appears in BAL due to account access (delegation target)
+                oracle: BalAccountExpectation(),
+            }
+        ),
+    )
+
+    post = {
+        alice: Account(
+            balance=10,
+            storage={0x02: 0x42},
+        ),
+        bob: Account(nonce=1),
+    }
+
+    blockchain_test(
+        pre=pre,
+        blocks=[block],
+        post=post,
+    )
+
+
+def test_bal_7702_invalid_nonce_authorization(
+    pre: Alloc,
+    blockchain_test: BlockchainTestFiller,
+):
+    """Ensure BAL handles failed authorization due to wrong nonce."""
+    alice = pre.fund_eoa()
+    bob = pre.fund_eoa(amount=0)
+    relayer = pre.fund_eoa()
+    oracle = pre.deploy_contract(code=Op.STOP)
+
+    tx = Transaction(
+        sender=relayer,  # Sponsored transaction
+        to=bob,
+        value=10,
+        gas_limit=1_000_000,
+        gas_price=0xA,
+        authorization_list=[
+            AuthorizationTuple(
+                address=oracle,
+                nonce=5,  # Wrong nonce - Alice's actual nonce is 0
+                signer=alice,
+            )
+        ],
+    )
+
+    block = Block(
+        txs=[tx],
+        expected_block_access_list=BlockAccessListExpectation(
+            account_expectations={
+                alice: BalAccountExpectation(
```

Comment:
BAL does not include authority when creating an delegation with an invalid nonce.